packages/@aws-cdk/aws-codedeploy/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,155 @@
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from './codedeploy.generated';
+
+export class DeploymentConfigurationName extends cdk.CloudFormationToken {}
+
+export class DeploymentConfigurationArn extends cdk.Arn {}
+
+/**
+ * Represents a Deployment Configuration for an EC2/on-premise Deployment Group.
+ */
+export interface ServerDeploymentConfiguration {
```

Comment:
This should be `IXxx` because it's not a data type.

Also, why do you need both the name and the ARN here? Are both needed downstream?

Reply:
> Also, why do you need both the name and the ARN here? Are both needed downstream?

Yep, both are needed at different spots.

> This should be IXxx because it's not a data type.

It's an interface because there are 2 kinds of Configs - the predefined ones (`OneAtATime`, `AllAtOnce`, etc.) and the custom ones (the ones you define with `minimumHealthyHosts`). Both implementations are data types. Naming this `IServerDeploymentConfiguration` seems incredibly ugly to me :-(. But if you think that's better, I can change it.

Reply:
As mentioned, our convention is that only data types (like "props" bags) can omit the `I` prefix. All other interfaces should be prefixed with `I`.